analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -577,3 +580,48 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
         }
     }
 }
+
+/**
+ * Several Maven repositories have disabled HTTP access and require HTTPS now. To be able to still analyze old Maven
+ * projects that use the HTTP URLs, this [MirrorSelector] implementation automatically creates an HTTPS mirror if a
+ * [RemoteRepository] uses a disabled HTTP URL. Without that Maven would abort with an exception as soon as it tries to
+ * download an Artifact from any of those repositories.
+ *
+ * **See also:**
+ *
+ * [GitHub Security Lab issue](https://github.com/github/security-lab/issues/21)
+ * [Medium article](https://medium.com/@jonathan.leitschuh/update-want-to-take-over-the-java-ecosystem-all-you-need-is-a-mitm-d069d253fe23)
+ */
+class HttpsMirrorSelector(private val originalMirrorSelector: MirrorSelector?) : MirrorSelector {
+    companion object {
+        private val DISABLED_HTTP_REPOSITORY_URLS = listOf(
+            "http://jcenter.bintray.com",
+            "http://repo.maven.apache.org",
+            "http://repo1.maven.org",
+            "http://repo.spring.io/"
+        )
+    }
+
+    override fun getMirror(repository: RemoteRepository?): RemoteRepository? {
+        originalMirrorSelector?.getMirror(repository)?.let { return it }
+
+        if (repository != null && repository.url in DISABLED_HTTP_REPOSITORY_URLS) {
+            log.info {
+                "HTTP access to ${repository.id} (${repository.url}) was disabled. Automatically switching to HTTPS."
+            }
+
+            return RemoteRepository.Builder(
+                "${repository.id}-ort-mirror",
+                repository.contentType,
+                "https://${repository.url.substringAfter("http://")}"
```

Comment:
Maybe use `removePrefix()` instead to emphasize we expect "http://" to be a prefix (and not in the middle of the string)?

Reply:
Done.